bertopic/_bertopic.py

```diff
@@ -478,23 +478,22 @@ def fit_transform(
         if documents.Document.values[0] is None:
             custom_documents = self._images_to_text(documents, embeddings)
 
-            # Extract topics by calculating c-TF-IDF
-            self._extract_topics(custom_documents, embeddings=embeddings)
-            self._create_topic_vectors(documents=documents, embeddings=embeddings)
-
-            # Reduce topics
+            # Reduce topics if needed, extract topics by calculating c-TF-IDF, and get representations.
             if self.nr_topics:
                 custom_documents = self._reduce_topics(custom_documents)
+            else:
+                self._extract_topics(custom_documents, embeddings=embeddings, verbose=self.verbose)
+            self._create_topic_vectors(documents=documents, embeddings=embeddings)
 
             # Save the top 3 most representative documents per topic
             self._save_representative_docs(custom_documents)
-        else:
-            # Extract topics by calculating c-TF-IDF
-            self._extract_topics(documents, embeddings=embeddings, verbose=self.verbose)
 
-            # Reduce topics
+        else:
+            # Reduce topics if needed, extract topics by calculating c-TF-IDF, and get representations.
             if self.nr_topics:
                 documents = self._reduce_topics(documents)
+            else:
+                self._extract_topics(documents, embeddings=embeddings, verbose=self.verbose)
```

Comment:
I'm missing something basic here... How is it possible that we can run `_reduce_topics` while no topic embeddings/ctfidf has been created already? Or has that been done somewhere and I'm completely glossing over it?